mono/mini/mini-x86.c

```diff
@@ -5425,7 +5425,8 @@ mono_arch_emit_epilog (MonoCompile *cfg)
 	if (need_stack_frame)
 		x86_leave (code);
 
-	if (CALLCONV_IS_STDCALL (sig)) {
+	if (CALLCONV_IS_STDCALL (sig) &&
+		method->wrapper_type != MONO_WRAPPER_MANAGED_TO_NATIVE) {
 		MonoJitArgumentInfo *arg_info = alloca (sizeof (MonoJitArgumentInfo) * (sig->param_count + 1));
```

Comment:
Hm, currently we don't change call_convention between the wrapper method and the eventual native call (in mono_marshal_emit_native_wrapper), so for a stdcall function, they must both be stdcall. However, we do set the signature's pinvoke field, and we check that in several other places. So perhaps the real problem is that CALLCONV_IS_STDCALL can be true for a non-pinvoke signature. Does that make sense to you?